src/main/java/seedu/address/logic/parser/ListCommandParser.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GUEST;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ROOM;
+
+import seedu.address.logic.commands.ListCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new ListCommand object
+ */
+
+public class ListCommandParser implements Parser<ListCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ListCommand
+     * and returns an ListCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListCommand parse(String args) throws ParseException {
+        //Splitting string and checking for exception
+        String[] splitString = args.trim().split("\\s+");
```

Comment:
Is this code a violation of SLAP? In fact, might want to move the `areFlagsPresent` to the `ParserUtil` class.

`ParserUtil` currently has a similar variant, `arePrefixesPresent`. Checking for flags would be the job of that class.

Reply:
Oops didnt see that, thanks!